test/cryptoFactory.test.js

```diff
@@ -80,4 +81,25 @@ describe("cryptoFactory and crypto methods", () => {
             expect(result).toEqual(output);
         });
     });
+
+    describe.each([
+        [["item1", "item2", "item3"], "item1\0item2\0item3"],
+        [["item1", "item2", "test1", "item2", "item3", "item2", "item3", "item2", "item3"], "item1\0item2\0test1\0item2\0item3\0item2\0item3\0item2\0item3"],
+        [["item1", "item2", "item3"], "item1\0item2\0item3"],
+        [["test"], "test"],
+        [[], ""]
+    ])("null_concat should produce correct output", (items, output) => {
+        
+        test("on nodejs", () => {
+            expect(crypto.null_concat.apply(this, items)).toEqual(output);
+        });
+
+        test("on browser", async () => {
+            const result = await page.evaluate(async (items) => {    
+                const cryptoLib = app.selectCrypto();
+                return cryptoLib.null_concat.apply(this, items);
+            }, items);
+            expect(result).toEqual(output);
+        });
```

Comment:
Because of the architecture that the browser tests use, puppeteer. I can't select the crypto version to use outside of the page evaluate function. As for where the `expect(result).toEqual(output)` line, I believe it created errors when it is inside the page evaluation call. I don't remember exactly though... I can put it in the list to look at. Does that answer your question?

Reply:
Yes, thanks